### ISSUE TITLE
Modify or replace order

### ISSUE THREAD

**Issue by walec51** [knowm/XChange#2818]
Some exchanges have API methods that allow to modify an order or cancel an existing one and create a new one in one request. This could be added to generic interfaces as one method:

```
Order replaceOrder(String currenctOrderId, Order newOrderData)
```

which will either modify an existing order or replace it depending on the exchanges API capabilities.

This feature was initially proposed in PR: https://github.com/knowm/XChange/pull/2806
however current API support seams not wide spread enough to add to generic interfaces.

Feature support in exchange APIs is monitored on: https://github.com/knowm/XChange/wiki/Generic-feature-candidates

**Comment by cymp** [comment]
I think it is a dangerous addition to the generic interface. This sure is a great feature when implemented, but unfortunately too rare. Will cause many client code to thrown unexpected exceptions.

**Comment by SergeyLukashevich** [comment]
> I think it is a dangerous addition to the generic interface. This sure is a great feature when implemented by the exchanges, but unfortunately too rare. 

There are already functions in the generic interface that hasn't implemented in particular exchanges. For instance `placeStopOrder`, `placeMarketOrder` and even `getOrder` aren't exists in 100% exchanges. There are no requirements that exchange have to realize all of them. Moreover, since all functions in interfaces are defined as `default`, I can presume that it's allowed by this library architecture. 

> Will cause many client codes to throw unexpected exceptions.

If developer will use library blindly, without looking at what features are supported, he gets exceptions even without `modifyOrder`.

**Comment by roeiba** [comment]
I agree with @SergeyLukashevich and I think it is OK to add such interface method, and add it to all supported exchanges and returning 'not implemented' exception on the exchanges that does not support replace feature.

I already have that interface at my code with implementation for Bitfinex and Poloniex only.

**Comment by walec51** [comment]
@SergeyLukashevich the fact that some generic features are not supported on 100% exchanges is not an argument to assume that features supported by a few exchanges are generic

some exchanges support `modifyOrder` and some support `replaceOrder`

so we could consider adding `String replaceOrModifyOrder(Order modifiedOrder)` which returns the new orders id or the unchanged id of the modified order depending on what capability is supported by the underlining exchange

however according to my research at https://github.com/knowm/XChange/wiki/Generic-feature-candidates this still does not cover > 50% of exchanges

so to not have a method that throws a NotYetImplementedException for  > 50% of exchanges we could add a default method to the interface like

```
String replaceOrModifyMarketOrder(MarketOrder modifiedOrder) {
    cancelOrder(modifiedOrder.getId());
    return placeMarketOrder(modifiedOrder);
}
```

**Comment by SergeyLukashevich** [comment]
Sounds good for me.  But `replaceOrModifyOrder` looks too long. How about `changeOrder`? it can be related to both modify and replace operations.

**Comment by walec51** [comment]
Ok for `changeOrder` but the javadoc in the interface should mention that on most exchanges this method will cancel the given order and create a new one in its place and on some it will modify an order preserving its id.
